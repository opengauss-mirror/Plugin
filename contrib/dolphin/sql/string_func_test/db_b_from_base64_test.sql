drop database if exists from_base64;
create database from_base64 dbcompatibility 'b';
\c from_base64

--测试正常base64编码作为输入，返回base64编码的解码结果
SELECT FROM_BASE64('YWJj');
SELECT FROM_BASE64('MTIzNDU2');
SELECT FROM_BASE64(TO_BASE64('测试'));

--测试非base64编码和NULL作为输入，均返回空值
SELECT FROM_BASE64('asjeifj');
SELECT FROM_BASE64(NULL);

--测试bit类型、bool类型、numeric类型、date类型作为输入
SELECT FROM_BASE64(b'01011001010101110100101001101010');
SELECT FROM_BASE64(TRUE);
SELECT FROM_BASE64(FALSE);
SELECT TO_BASE64(FROM_BASE64(123456781234567812345678));
SELECT TO_BASE64(FROM_BASE64('123456781234567812345678'));
SELECT FROM_BASE64(now());

-- 测试table类型作为输入
CREATE TABLE test_base64 (name text);
INSERT INTO test_base64 values('YWJj'), ('MTIzNDU2'), ('asjeifj');
SELECT FROM_BASE64(name) from test_base64;

\c postgres
drop database if exists from_base64;