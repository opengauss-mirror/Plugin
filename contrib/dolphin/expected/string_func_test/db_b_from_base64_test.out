create schema from_base64;
set current_schema to 'from_base64';
--测试正常base64编码作为输入，返回base64编码的解码结果
SELECT FROM_BASE64('YWJj');
 from_base64 
-------------
 abc
(1 row)

SELECT FROM_BASE64('MTIzNDU2');
 from_base64 
-------------
 123456
(1 row)

SELECT FROM_BASE64(TO_BASE64('测试'));
 from_base64 
-------------
 测试
(1 row)

--测试非base64编码和NULL作为输入，均返回空值
SELECT FROM_BASE64('asjeifj');
 from_base64 
-------------
 
(1 row)

SELECT FROM_BASE64(NULL);
 from_base64 
-------------
 
(1 row)

--测试bit类型、bool类型、numeric类型、date类型作为输入
SELECT FROM_BASE64(b'01011001010101110100101001101010');
 from_base64 
-------------
 abc
(1 row)

SELECT FROM_BASE64(TRUE);
 from_base64 
-------------
 
(1 row)

SELECT FROM_BASE64(FALSE);
 from_base64 
-------------
 
(1 row)

SELECT TO_BASE64(FROM_BASE64(123456781234567812345678));
        to_base64         
--------------------------
 123456781234567812345678
(1 row)

SELECT TO_BASE64(FROM_BASE64('123456781234567812345678'));
        to_base64         
--------------------------
 123456781234567812345678
(1 row)

SELECT FROM_BASE64(now());
 from_base64 
-------------
 
(1 row)

-- 测试table类型作为输入
CREATE TABLE test_base64 (name text);
INSERT INTO test_base64 values('YWJj'), ('MTIzNDU2'), ('asjeifj');
SELECT FROM_BASE64(name) from test_base64;
 from_base64 
-------------
 abc
 123456
 
(3 rows)

drop schema from_base64 cascade;
NOTICE:  drop cascades to table test_base64
reset current_schema;
